src/overlays/actors/ovl_Obj_Driftice/z_obj_driftice.c

```diff
@@ -325,129 +393,131 @@ void ObjDriftice_Destroy(Actor* thisx, PlayState* play) {
     DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
 }
 
-void func_80A671A8(ObjDriftice* this) {
-    this->actionFunc = func_80A671BC;
+void ObjDriftice_SetupDoNothing(ObjDriftice* this) {
+    this->actionFunc = ObjDriftice_DoNothing;
 }
 
-void func_80A671BC(ObjDriftice* this, PlayState* play) {
+void ObjDriftice_DoNothing(ObjDriftice* this, PlayState* play) {
 }
 
-void func_80A671CC(ObjDriftice* this) {
-    this->actionFunc = func_80A671E0;
+void ObjDriftice_SetupDriftAlongPath(ObjDriftice* this) {
+    this->actionFunc = ObjDriftice_DriftAlongPath;
 }
 
-void func_80A671E0(ObjDriftice* this, PlayState* play) {
-    f32 phi_f0;
-    Vec3f sp40;
-    f32 sp3C;
-    f32 phi_f12;
-    Vec3s* points;
-    s32 sp30;
+void ObjDriftice_DriftAlongPath(ObjDriftice* this, PlayState* play) {
+    f32 speedTarget;
+    Vec3f nextPathPoint;
+    f32 distToNextPathPoint;
+    f32 speedStep;
+    Vec3s* finalPathPoint;
+    s32 snapToPathPoint;
     Actor* thisx = &this->dyna.actor;
 
-    Math_Vec3s_ToVec3f(&sp40, &(&this->pathPoints[this->unk_164])[this->unk_168]);
-    Math_Vec3f_Diff(&sp40, &this->dyna.actor.world.pos, &thisx->velocity);
+    Math_Vec3s_ToVec3f(&nextPathPoint, &(&this->pathPoints[this->curPathPointIdx])[this->pathDirection]);
+    Math_Vec3f_Diff(&nextPathPoint, &this->dyna.actor.world.pos, &thisx->velocity);
 
-    sp3C = Math3D_Vec3fMagnitude(&thisx->velocity);
-    if (sp3C < (this->unk_23C * 8.0f)) {
-        phi_f0 = this->unk_23C * 0.4f;
-        phi_f12 = this->unk_23C * 0.05f;
+    distToNextPathPoint = Math3D_Vec3fMagnitude(&thisx->velocity);
+    if (distToNextPathPoint < (this->driftSpeed * 8.0f)) {
+        speedTarget = this->driftSpeed * 0.4f;
+        speedStep = this->driftSpeed * 0.05f;
     } else {
-        phi_f0 = this->unk_23C;
-        phi_f12 = this->unk_23C * 0.13f;
+        speedTarget = this->driftSpeed;
+        speedStep = this->driftSpeed * 0.13f;
     }
 
-    Math_StepToF(&this->dyna.actor.speed, phi_f0, phi_f12);
+    Math_StepToF(&this->dyna.actor.speed, speedTarget, speedStep);
 
-    if ((this->dyna.actor.speed + 0.05f) < sp3C) {
-        Math_Vec3f_Scale(&this->dyna.actor.velocity, this->dyna.actor.speed / sp3C);
+    if ((this->dyna.actor.speed + 0.05f) < distToNextPathPoint) {
+        Math_Vec3f_Scale(&this->dyna.actor.velocity, this->dyna.actor.speed / distToNextPathPoint);
         this->dyna.actor.world.pos.x += this->dyna.actor.velocity.x;
         this->dyna.actor.world.pos.y += this->dyna.actor.velocity.y;
         this->dyna.actor.world.pos.z += this->dyna.actor.velocity.z;
     } else {
-        sp30 = true;
-        this->unk_164 += this->unk_168;
+        snapToPathPoint = true;
+        this->curPathPointIdx += this->pathDirection;
 
         if (1) {}
```

Comment:
If you already tried to remove this `if (1)` with the permuter, ignore this comment. Otherwise, can you see if it can be removed with the permuter? Don't waste too much time on this if it doesn't work quickly as it's not abnormal to have to resort to that for matching.